corehq/apps/locations/static/locations/js/widgets.js

```diff
@@ -16,7 +16,12 @@ hqDefine("locations/js/widgets", [
         _.each($source.select2('data'), function (result) {
             const fullLengthName = result.text || result.name;
             const truncatedName = truncateLocationName(fullLengthName, $select);
-            var $option = new Option(truncatedName, result.id);
+            let $option;
+            if (toggles.toggleEnabled('LOCATION_FIELD_USER_PROVISIONING')) {
+                $option = new Option(truncatedName, result.id);
```

Comment:
nit: `truncatedName` initialization can go in this conditional too

Reply:
have to admit I liked the kind of symmetry of having both vars together then choosing which to use, but agree it's better to move `truncatedName`.
made the change here: https://github.com/dimagi/commcare-hq/pull/35072/commits/d84f3cc3b97b0fcad2c82668fded54018535c593

Reply:
I guess the same argument probably goes for the other call to `truncateLocationName` too